src/observer.js

```diff
@@ -367,16 +367,17 @@ export function observer(arg1, arg2) {
         throw new Error("Please pass a valid component to 'observer'")
     }
 
-    const target = componentClass.prototype || componentClass
+    const targetClass = class targetClassName extends componentClass {}
```

Comment:
Wondering, will `targetClassName` now be the reported name in exceptions and React devtools and such?

Reply:
It will, so it's worth implementing `.displayName`

Reply:
Yeah, just added `displayName` for `targetClass`